internal/grader/grade.go

```diff
@@ -18,19 +18,21 @@ var submissionLocks sync.Map
 type GradeOptions struct {
 	NoDocker     bool
 	LeaveTempDir bool
+	AllowLate    bool
 }
 
-func GetDefaultGradeOptions() GradeOptions {
+func GetDefaultGradeOptions(allowLate bool) GradeOptions {
```

Comment:
This argument breaks the pattern.
(It's not much of a default if you have to select an option.)

Instead, callers should get the default option and then set AllowLate if need be.